### ISSUE TITLE
Thumbnail preview of problems desired

### ISSUE THREAD

**Issue by mekersh** [PrairieLearn/PrairieLearn#1731]
It's cumbersome to have to click on each question within a given assessment. Could we create a thumbnail view of the image used in the question (if applicable) and have an option to view the assessment's thumbnails?

**Comment by nwalters512** [comment]
Before we get into the technical bits that would make this so tricky, I want to dig a bit deeper into why you want this. Can you elaborate on "it's cumbersome to have to click on each question within a given assessment"? To answer a question, you'll naturally have to click through to it; adding a thumbnail image doesn't fix that. Is this a problem for you when you first open an assessment and want to quickly get a sense of what questions exist, or when you're trying to remember which questions to revisit while taking the exam, or something else entirely?

**Comment by mekersh** [comment]
I’d like a way to look at an assessment and get an idea of what questions are on it.  This is more for the process of developing assessments.  For example, the names/qid aren’t helpful and as we add more questions to the database, it's hard to keep track.  If I could look at the question list and see a thumbnail, this would be the most helpful.

**Comment by nwalters512** [comment]
Gotcha, thanks for the context that you're on the course side of things. Could you elaborate on why question names aren't as helpful? QIDs will naturally be more opaque, but question names are meant to help both instructors and students quickly gauge the contents of each question. Are you and other instructors having difficulty coming up with descriptive names?

**Comment by mwest1066** [comment]
I think that in highly graphical subjects (like mechanics) people often describe questions by what they “look like”. I know I tend to say “it’s the question with a ladder leaning on a wall” or something like that. I must admit that I would like to have thumbnails too 😄

A straightforward implementation for thumbnails would be to look for `thumbnail.{jpg,png}` in the question directory at sync time, and to store this filename in the `questions` table if present. The we can embed this image into instructor question lists. Alternatively, we could require `”thumbnail”: “thumbnail.jpg”` in the `info.json`.

**Comment by mekersh** [comment]
Yes, exactly as Matt said.  For me it’s easier to see the picture and know what the question type is and I can then gauge if we have a good distribution, modify, etc.


On Oct 3, 2019, at 11:44 AM, Matthew West <notifications@github.com<mailto:notifications@github.com>> wrote:


I think that in highly graphical subjects (like mechanics) people often describe questions by what they “look like”. I know I tend to say “it’s the question with a ladder leaning on a wall” or something like that. I must admit that I would like to have thumbnails too 😄

A straightforward implementation for thumbnails would be to look for thumbnail.{jpg,png} in the question directory at sync time, and to store this filename in the questions table if present. The we can embed this image into instructor question lists. Alternatively, we could require ”thumbnail”: “thumbnail.jpg” in the info.json.

**Comment by trombonekenny** [comment]
Wayne Chang is sponsoring students in SP2020 to implement this feature.

**Comment by reteps** [comment]
This came up recently when discussing the assessment editor.

**Comment by mwest1066** [comment]
I still think it's a good idea. Not that I have any simple solution for implementation.
